src/Notifier.cpp

```diff
@@ -61,6 +63,24 @@ bool Notifier::Initialize(const size_t handshakeCount) {
 }
 
 /*
+ * Receives message data without caring about handshakes, added sleeping code
+ * to prevent pounding the cpu
+ */
+std::string Notifier::ReceiveData() {
+   StopWatch waitCheck;
+   while (waitCheck.ElapsedSec() < mMaxTimeoutInSec && mKeepRunning.load()) {
```

Comment:
I think we don't need ```mKeepRunning```. Usually we've used ```atomic<bool>``` for threaded tests so that the variable can be changed from outside the thread. The Notifier is not it's own thread however. Really, this loop just needs to run until the timeout, or until ```msg != ""```